ES integration test is flakey

<!--
Welcome to the Jaeger project! 👋🎉

- Please search for existing issues to avoid creating duplicate bugs/feature requests.
- Please be respectful and considerate of others when commenting on issues.
- Please provide as much information as possible so we all understand the issue.
- If you only have a question, you may get a faster response by asking in
    - our chat room https://gitter.im/jaegertracing/Lobby, or
    - the forum https://groups.google.com/d/forum/jaeger-tracing
    (but please don't double post)
-->

## Requirement - what kind of business use case are you trying to solve?
Have tests be indicative of real problems
<!-- required section -->

## Problem - what in Jaeger blocks you from solving the requirement?
```
=== RUN   TestElasticsearchStorage/GetDependencies
--- FAIL: TestElasticsearchStorage (89.25s)
    --- FAIL: TestElasticsearchStorage/GetTrace (4.21s)
        integration_test.go:100: Waiting for storage backend to update documents, iteration 1 out of 30
. to update documents, iteration 30 out of 30
        integration_test.go:181: No trace with that ID found
        integration_test.go:181: No trace with that ID found
        <autogenerated>:1: 
            	Error Trace:	integration_test.go:185
            	            				integration_test.go:378
            	Error:      	Should be true
            	Test:       	TestElasticsearchStorage/GetTrace
        <autogenerated>:1: 
            	Error Trace:	domain_trace_compare_test.go:53
            	            				integration_test.go:186
            	            				integration_test.go:378
            	Error:      	Expected value not to be nil.
            	Test:       	TestElasticsearchStorage/GetTrace
    --- FAIL: TestElasticsearchStorage/GetLargeSpans (78.75s)
        integration_test.go:132: Testing Large Trace over 10K ...
        integration_test.go:100: Waiting for storage backend to update documents, iteration 1 out of 30
.
        integration_test.go:100: Waiting for storage backend to update documents, iteration 30 out of 30
        <autogenerated>:1: 
            	Error Trace:	integration_test.go:143
            	            				integration_test.go:379
            	Error:      	Should be true
            	Test:       	TestElasticsearchStorage/GetLargeSpans
        <autogenerated>:1: 
            	Error Trace:	domain_trace_compare_test.go:53
            	            				integration_test.go:144
            	            				integration_test.go:379
            	Error:      	Expected value not to be nil.
            	Test:       	TestElasticsearchStorage/GetLargeSpans
```

## Proposal - what do you suggest to solve the problem or improve the existing situation?

TBD

--- a/scripts/travis/es-integration-test.sh
+++ b/scripts/travis/es-integration-test.sh
@@ -2,8 +2,8 @@
 
 set -e
 
-docker pull docker.elastic.co/elasticsearch/elasticsearch:5.4.0
-CID=$(docker run -d -p 9200:9200 -e "http.host=0.0.0.0" -e "transport.host=127.0.0.1" docker.elastic.co/elasticsearch/elasticsearch:5.4.0)
+docker pull docker.elastic.co/elasticsearch/elasticsearch:5.6.12
+CID=$(docker run -d -p 9200:9200 -e "http.host=0.0.0.0" -e "transport.host=127.0.0.1" docker.elastic.co/elasticsearch/elasticsearch:5.6.12)
 export STORAGE=elasticsearch
 make storage-integration-test
 docker kill $CID

